sdk/servicebus/Azure.Messaging.ServiceBus/src/Compatibility/ServiceBusAmqpExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Messaging.ServiceBus.Amqp;
+using Microsoft.Azure.Amqp;
+using AmqpLib = Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.ServiceBus.Primitives
+{
+    /// <summary>
+    /// Extensions for converting to and from AMQP.
+    /// </summary>
+    public static class ServiceBusAmqpExtensions
+    {
+        /// <summary>
+        /// Converts the <see cref="ServiceBusReceivedMessage"/> to its serialized AMQP form.
+        /// </summary>
+        /// <returns></returns>
+        public static BinaryData ToAmqpBytes(this ServiceBusReceivedMessage message)
+        {
+            var stream = AmqpMessageConverter.Default.AmqpAnnotatedMessageToAmqpMessage(message.GetRawAmqpMessage()).ToStream();
+            return BinaryData.FromStream(stream);
```

Comment:
isn't this converting byte[] to stream, and then stream to bytes[]?

Reply:
Yes, but the format we need is BinaryData, and the AMQP library doesn't currently expose a way to get the bytes directly.

Reply:
We might want to ask the AMQP team to add an API for it.

Reply:
Filed https://github.com/Azure/azure-amqp/issues/238